aws-inventory/lambda/inventory-transit-gateway.py

```diff
@@ -0,0 +1,155 @@
+
+
+def lambda_handler(event, context):
+    logger.debug("Received event: " + json.dumps(event, sort_keys=True))
+    message = json.loads(event['Records'][0]['Sns']['Message'])
+    logger.info("Received message: " + json.dumps(message, sort_keys=True))
+
+    try:
+        target_account = AWSAccount(message['account_id'])
+            
+        for r in target_account.get_regions():
+            try:
+                discover_transit_gateways(target_account, r)
+            except ClientError as e:
+                # Move onto next region if we get access denied. This is probably SCPs
+                if e.response['Error']['Code'] == 'AccessDeniedException':
+                    logger.error(f"AccessDeniedException for region {r} in function {context.function_name} for {target_account.account_name}({target_account.account_id})")
+                    continue
+                else:
+                    raise  # pass on to the next handler
+
+    except AntiopeAssumeRoleError as e:
+        logger.error("Unable to assume role into account {}({})".format(target_account.account_name, target_account.account_id))
+        return()
+    except ClientError as e:
+        if e.response['Error']['Code'] == 'UnauthorizedOperation':
+            logger.error("Antiope doesn't have proper permissions to this account")
+            return(event)
+        logger.critical("AWS Error getting info for {}: {}".format(message['account_id'], e))
+        capture_error(message, context, e, "ClientError for {}: {}".format(message['account_id'], e))
+        raise
+    except Exception as e:
+        logger.critical("{}\nMessage: {}\nContext: {}".format(e, message, vars(context)))
+        capture_error(message, context, e, "General Exception for {}: {}".format(message['account_id'], e))
+        raise
+
+def discover_transit_gateways(target_account, region):
+    '''Iterate accross all regions to discover transit gateways'''
+    
+    ec2_client = target_account.get_client('ec2', region=region)
+    response = ec2_client.describe_transit_gateways()
+
+    if response['TransitGateways']:
+    
+        for tg in response['TransitGateways']:
+        
+            resource_item = {}
+            resource_item['awsAccountId']                   = target_account.account_id
+            resource_item['awsAccountName']                 = target_account.account_name
+            resource_item['resourceType']                   = "AWS::EC2::TransitGateway"
+            resource_item['source']                         = "Antiope"
+            resource_item['awsRegion']                      = region
+            resource_item['configurationItemCaptureTime']   = str(datetime.datetime.now())
+            resource_item['configuration']                  = tg
+            resource_item['supplementaryConfiguration']     = {}
+            resource_item['resourceId']                     = tg['TransitGatewayId']
+            resource_item['errors']                         = {}
```

Comment:
Can you add the resource_item['ARN'] mapped to tg['TransitGatewayArn']?

Reply:
You can also map resource_item['resourceCreationTime']        to tg['CreationTime']